test/functional/wallet_listsinceblock.py

```diff
@@ -74,6 +75,27 @@ def test_invalid_blockhash(self):
         assert_raises_rpc_error(-8, "blockhash must be hexadecimal string (not 'Z000000000000000000000000000000000000000000000000000000000000000')", self.nodes[0].listsinceblock,
                                 "Z000000000000000000000000000000000000000000000000000000000000000")
 
+    def test_targetconfirmations(self):
+        '''
+        This tests when the value of target_confirmations exceeds the number of
+        blocks in the main chain. In this case, the genesis block hash should be
+        given for the `lastblock` property. If target_confirmations is < 1, then
+        a -8 invalid parameter error is thrown.
+        '''
+        self.log.info("Test target_confirmations")
+        blockhash, = self.nodes[2].generate(1)
+        blockheight = self.nodes[2].getblockheader(blockhash)['height']
+        self.sync_all()
+
+        assert_equal(
+            self.nodes[0].getblockhash(0),
+            self.nodes[0].listsinceblock(blockhash, blockheight + 1)['lastblock'])
+        assert_equal(
+            self.nodes[0].getblockhash(0),
+            self.nodes[0].listsinceblock(blockhash, blockheight + 2)['lastblock'])
```

Comment:
In commit "Return genesis hash" (c0e57fe0d399235f3f2612c4972c597b34107509)

Maybe test with a larger value like 1000 to be sure this isn't an edge case